### PR TITLE
examples: add a couple of @playwright/test examples

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: examples/todomvc/tests/integration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+const TODO_ITEMS = [
+  'buy some cheese',
+  'feed the cat',
+  'book a doctors appointment'
+];
+
+test.describe('New Todo', () => {
+  test('should allow me to add todo items', async ({ page }) => {
+    // Create 1st todo.
+    await page.locator('.new-todo').fill(`${TODO_ITEMS[0]}`);
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:18]:
```suggestion
    await page.locator('.new-todo').fill(TODO_ITEMS[0]);
```

File: examples/todomvc/tests/integration.spec.ts [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+    await page.locator('.new-todo').press(`Enter`);
+
+    // Make sure the list only has one todo item.
+    await expect(page.locator('.view label')).toHaveText([
+      TODO_ITEMS[0]
+    ]);
+
+    // Create 2nd todo.
+    await page.locator('.new-todo').fill(`${TODO_ITEMS[1]}`);
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:27]:
```suggestion
    await page.locator('.new-todo').fill(TODO_ITEMS[1]);
```

File: examples/todomvc/tests/integration.spec.ts [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+    await expect(page.locator('.todo-count')).toContainText('3');
+    await expect(page.locator('.todo-count')).toHaveText(/3/);
+
+    // Check all items in one call.
+    await expect(page.locator('.view label')).toHaveText(TODO_ITEMS);
+    await checkNumberOfTodosInLocalStorage(page, 3);
+  });
+
+  test('should show #main and #footer when items added', async ({ page }) => {
+    await page.locator('.new-todo').fill(`${TODO_ITEMS[0]}`);
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:64]:
```suggestion
    await page.locator('.new-todo').fill(TODO_ITEMS[0]);
```

File: examples/todomvc/tests/integration.spec.ts [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+
+  test.afterEach(async ({ page }) => {
+    await checkNumberOfTodosInLocalStorage(page, 3);
+  });
+
+  test('should allow me to mark all items as completed', async ({ page }) => {
+    // Complete all todos.
+    await page.locator('.toggle-all').check();
+
+    // Get each all todo lis and ensure they all have 'completed' class.
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:87]:
```suggestion
    // Ensure all todos have 'completed' class.
```

File: examples/todomvc/tests/integration.spec.ts [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+    await checkNumberOfCompletedTodosInLocalStorage(page, 3);
+
+    // Uncheck first todo.
+    const firstTodo = page.locator('.todo-list li').nth(0);
+    await firstTodo.locator('.toggle').uncheck();
+
+    // Reuse toggleAll locator and make sure its not checked.
+    await expect(toggleAll).not.toBeChecked();
+
+    // Reuse toggleAll locator and now toggle it.
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:114]: The comment is out of sync with the code.

File: examples/todomvc/tests/integration.spec.ts [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+    await expect(toggleAll).toBeChecked();
+  });
+});
+
+test.describe('Item', () => {
+
+  test('should allow me to mark items as complete', async ({ page }) => {
+    // Create two items.
+    for (const item of TODO_ITEMS.slice(0, 2)) {
+      await page.locator('.new-todo').fill(`${item}`);
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:129]:
```suggestion
      await page.locator('.new-todo').fill(item);
```

File: examples/todomvc/tests/integration.spec.ts [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+
+    // Assert completed class.
+    await expect(firstTodo).toHaveClass('completed');
+    await expect(secondTodo).toHaveClass('completed');
+  });
+
+  test('should allow me to un-mark items as complete', async ({ page }) => {
+    // Create two items.
+    for (const item of TODO_ITEMS.slice(0, 2)) {
+      await page.locator('.new-todo').fill(`${item}`);
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:151]:
```suggestion
      await page.locator('.new-todo').fill(item);
```

File: examples/svgomg/package.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "svgomg-tests",
+  "version": "0.0.1",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "npx playwright test",
```
Comment (mxschmitt) [inline examples/svgomg/package.json:7]: nit: `npx` is not needed there. Removing it makes it also faster.

File: examples/svgomg/package.json [hunk 9]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "svgomg-tests",
+  "version": "0.0.1",
+  "description": "",
+  "main": "index.js",
```
Comment (mxschmitt) [inline examples/svgomg/package.json:5]:
```suggestion
```

File: examples/svgomg/tests/example.spec.ts [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+      'Show original',
+      'Compare gzipped',
+      'Prettify markup',
+      'Multipass',
+    ]);
+
+    const toggle = page.locator('.setting-item-toggle');
+    await expect(toggle.locator('text=Show original')).not.toBeChecked();
+    await expect(toggle.locator('text=Compare gzipped')).toBeChecked();
+    await page.pause();
```
Comment (mxschmitt) [inline examples/svgomg/tests/example.spec.ts:36]: this should not be there I guess.

File: examples/todomvc/tests/integration.spec.ts [hunk 11]
```diff
@@ -0,0 +1,403 @@
...
+    const todoItems = page.locator('.todo-list li');
+    await todoItems.nth(1).dblclick();
+    await todoItems.nth(1).locator('.edit').press('Escape');
+    await expect(todoItems).toHaveText(TODO_ITEMS);
+  });
+});
+
+test.describe('Counter', () => {
+  test('should display the current number of todo items', async ({ page }) => {
+    await page.locator('.new-todo').fill(`${TODO_ITEMS[0]}`);
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:253]:
```suggestion
    await page.locator('.new-todo').fill(TODO_ITEMS[0]);
```

File: examples/todomvc/tests/integration.spec.ts [hunk 12]
```diff
@@ -0,0 +1,403 @@
...
+    await page.locator('.new-todo').press(`Enter`);
+    await expect(page.locator('.todo-count')).toContainText('1');
+
+    await page.locator('.new-todo').fill(`${TODO_ITEMS[1]}`);
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:257]:
```suggestion
    await page.locator('.new-todo').fill(TODO_ITEMS[1]);
```

File: examples/todomvc/tests/integration.spec.ts [hunk 13]
```diff
@@ -0,0 +1,403 @@
...
+    await page.locator('.todo-list li .toggle').first().check();
+    await page.locator('.clear-completed').click();
+    await expect(page.locator('.clear-completed')).toBeHidden();
+  });
+});
+
+test.describe('Persistence', () => {
+  test('should persist its data', async ({ page }) => {
+    for (const item of TODO_ITEMS.slice(0, 2)) {
+      await page.locator('.new-todo').fill(`${item}`);
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:293]:
```suggestion
      await page.locator('.new-todo').fill(item);
```

**Review by mxschmitt — APPROVED** [review]

File: examples/todomvc/tests/integration.spec.ts [hunk 14]
```diff
@@ -0,0 +1,403 @@
...
+  });
+});
+
+test.describe('Routing', () => {
+  test.beforeEach(async ({ page }) => {
+    await createDefaultTodos(page);
+    // make sure the app had a chance to save updated todos in storage
+    // before navigating to a new view, otherwise the items can get lost :(
+    // in some frameworks like Durandal
+    checkTodosInLocalStorage(page, TODO_ITEMS[0]);
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:318]: Missing await?

File: examples/todomvc/tests/integration.spec.ts [hunk 15]
```diff
@@ -0,0 +1,403 @@
...
+  });
+
+  test('should allow me to display active items', async ({ page }) => {
+    await page.locator('.todo-list li .toggle').nth(1).check();
+    await checkNumberOfCompletedTodosInLocalStorage(page, 1);
+    await page.locator('.filters li a >> text=Active').click();
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:324]: Perhaps just `'.filters >> text=Active'`?

File: examples/todomvc/tests/integration.spec.ts [hunk 16]
```diff
@@ -0,0 +1,403 @@
...
+    await expect(page.locator('.filters >> text=Active')).toHaveClass('selected');
+    await page.locator('.filters >> text=Completed').click();
+    // Page change - completed items.
+    await expect(page.locator('.filters >> text=Completed')).toHaveClass('selected');
+  });
+});
+
+async function createDefaultTodos(page: Page) {
+  for (const item of TODO_ITEMS) {
+    await page.locator('.new-todo').fill(`${item}`);
```
Comment (dgozman) [inline examples/todomvc/tests/integration.spec.ts:382]:
```suggestion
    await page.locator('.new-todo').fill(item);
```

**Review by dgozman — APPROVED** [review]

**Pull request merged**